Runtime/Components/CinemachineSplineDolly.cs

```diff
@@ -0,0 +1,368 @@
+            /// </summary>
+            [HideInInspector]
+            [Tooltip("The nearest point is calculated by finding the nearest point on the entire length of the spline " +
+                "using SearchResolution to divide into equally spaced line segments. Successive iterations will then " +
+                "subdivide further the nearest segment, producing more accurate results. In most cases, the default value (2) is sufficient.")]
+            public int m_SearchIteration;
+        }
+
+        /// <summary>Controls how automatic dollying occurs</summary>
+        [Tooltip("Controls how automatic dollying occurs.  A Follow target is necessary to use this feature.")]
+        public AutoDolly m_AutoDolly = new AutoDolly 
+        { 
+            m_Enabled = false, 
+            m_PositionOffset = 0,
+            m_SearchIteration = 4, 
+            m_SearchResolution = 2,
+        };
+
+        /// <summary>True if component is enabled and has a spline</summary>
+        public override bool IsValid => enabled && m_Spline != null;
+
+        /// <summary>Get the Cinemachine Pipeline stage that this component implements.
+        /// Always returns the Body stage</summary>
+        public override CinemachineCore.Stage Stage => CinemachineCore.Stage.Body;
+
+        /// <summary>
+        /// Report maximum damping time needed for this component.
+        /// </summary>
+        /// <returns>Highest damping setting in this component</returns>
+        public override float GetMaxDampTime() => 
+            Mathf.Max(Mathf.Max(m_Damping.x, Mathf.Max(m_Damping.y, m_Damping.z)), m_AngularDamping);
+
+        /// <summary>
+        /// Subscribe to onSplineChanged if you'd like to react to changes to the Spline attached to this vcam.
+        /// This action is invoked by the Spline's changed event when a spline property is modified. Available in editor only.
+        /// </summary>
+        public event Action onSplineChanged;
+
+        bool m_Registered = false;
+        SplineContainer m_SplineCache;
+        void OnValidate()
+        {
+            if (m_SplineCache != null)
+            {
+                m_SplineCache.Spline.changed -= onSplineChanged;
+                m_SplineCache = m_Spline;
+                m_Registered = false;
+            }
+            if (!m_Registered && m_Spline != null && m_Spline.Spline != null)
+            {
+                m_Registered = true;
+                m_SplineCache = m_Spline;
+                m_Spline.Spline.changed += onSplineChanged;
+            }
+
+            m_Damping.x = Mathf.Clamp(m_Damping.x, 0, 20);
+            m_Damping.y = Mathf.Clamp(m_Damping.y, 0, 20);
+            m_Damping.z = Mathf.Clamp(m_Damping.z, 0, 20);
+            m_AngularDamping = Mathf.Clamp(m_AngularDamping, 0, 20);
+        }
+
+        CinemachineSplineRoll m_Roll; // don't use this directly
+        CinemachineSplineRoll SplineRoll
```

Comment:
If no spline roll is present, this will keep trying to fetch it every frame from 2 different sources.  Is this a performance issue?  How expensive is TryGetComponent?

Reply:
Yes, that's not so good.
I checked in the profiler and it takes 0.02 for this call, which seems a lot. If I'd just return null, then it takes 0.00.

A potential solution would be:
- CmSplineRoll knows about vcams. Ping vcam, when CmSplineRoll is added to a vcam.
- CmSplineRoll knows about SplineContainer. When CmSplineRoll is added to a SplineContainer, it triggers an onSplineChanged, to which SplineDolly is registered. When this is called, we update m_Roll.

This way, we won't need to poll.

We had something like this in a previous iteration, but we did not like it.

Reply:
Why don't you just cache it in Awake?  Only attempt a TryGetComponent if editor and not playing.